scripts/upgrade

```diff
@@ -1,141 +1,216 @@
-SYSTEM_PY_VERSION=$(readlink /usr/bin/python3|sed s/.*python//)
+domain=$(ynh_app_setting_get --app=$app --key=domain)
+path_url=$(ynh_app_setting_get --app=$app --key=path)
+final_path=$(ynh_app_setting_get --app=$app --key=final_path)
+db_name=$(ynh_app_setting_get --app=$app --key=db_name)
+db_user=$db_name
+db_pwd=$(ynh_app_setting_get --app=$app --key=mysqlpwd)
+secret_key=$(ynh_app_setting_get --app=$app --key=secret_key)
+
+#=================================================
+# CHECK VERSION
+#=================================================
+
+### This helper will compare the version of the currently installed app and the version of the upstream package.
+### $upgrade_type can have 2 different values
+### - UPGRADE_APP if the upstream app version has changed
+### - UPGRADE_PACKAGE if only the YunoHost package has changed
+### ynh_check_app_version_changed will stop the upgrade if the app is up to date.
+### UPGRADE_APP should be used to upgrade the core app only if there's an upgrade to do.
+upgrade_type=$(ynh_check_app_version_changed)
+
+if ynh_compare_current_package_version --comparison le --version "4.1.5~ynh2"; then
+	upgrade_from_opt=true
+else
+	upgrade_from_opt=false
+fi
 
-# Source local utils
-source _common.sh
+#=================================================
+# BACKUP BEFORE UPGRADE THEN ACTIVE TRAP
+#=================================================
+ynh_script_progression --message="Backing up the app before upgrading (may take a while)..." --weight=3
 
+# Backup the current version of the app
+ynh_backup_before_upgrade
 ynh_clean_setup () {
-    if [ -e /opt/yunohost/ihatemoney/venv-old ]
-    then
-        mv /opt/yunohost/ihatemoney/venv{-old,}
-    fi
+	# Restore it if the upgrade fails
+	ynh_restore_upgradebackup
 }
-
+# Exit if an error occurs during the execution of the script
 ynh_abort_if_errors
 
+#=================================================
+# STANDARD UPGRADE STEPS
+#=================================================
+# STOP SYSTEMD SERVICE
+#=================================================
 
-#----------------------------PRE-UPGRADE MIGRATIONS-----------------------
-
-
+if [[ "$upgrade_from_opt" == "false" ]]; then
+	ynh_script_progression --message="Stopping a systemd service..." --weight=1
 
-# MIGRATION: upgrade arg to typed boolean form
-
-if (($is_public != 0)) && (($is_public != 1))
-then
-    if [ $is_public = "No" ];
-    then
-        is_public=0
-    else
-        is_public=1
-    fi
-    ynh_app_setting_set "$app" is_public "$is_public"
+	ynh_systemd_action --service_name=$app --action="stop" --log_path=systemd
 fi
 
+#=================================================
+# ENSURE DOWNWARD COMPATIBILITY
+#=================================================
+ynh_script_progression --message="Ensuring downward compatibility..." --weight=1
 
+# Cleaning legacy permissions
+if ynh_legacy_permissions_exists; then
+	ynh_legacy_permissions_delete_all
 
-# MIGRATION: Switch to a python3 venv
```

Comment:
I'd like to keep that as well for not breaking upgrade paths from old versions.

Reply:
It's useless as the install directory completely changed.